CMakeLists.txt

```diff
@@ -165,34 +165,33 @@ IF (_HAVE_LINK_H)
   ENDFOREACH()
 ENDIF()
 
+DEAL_II_INVOKE_AUTOPILOT()
+
 
+# Check if we can raise floating point exceptions.
+#
+# Note that some library we link with in ASPECT on some platforms will trigger
+# floating point exceptions when converting -numeric_limits<double>::max to a
+# string. The only thing we can do is a configure check and disable the
+# exceptions. This is done here:
 SET(ASPECT_USE_FP_EXCEPTIONS ON CACHE BOOL "If ON, floating point exception are raised in debug mode.")
 
-IF (ASPECT_USE_FP_EXCEPTIONS 
-    AND CMAKE_CXX_COMPILER_ID MATCHES "Intel"
-    AND CMAKE_CXX_COMPILER_VERSION VERSION_LESS "15"
-    )
-  # deal.II is triggering floating point exception in Pattern::Double() so disable them
-  MESSAGE("Runtime floating point checks not supported on Intel compiler. Disabling...")
-  SET(ASPECT_USE_FP_EXCEPTIONS OFF CACHE BOOL "" FORCE)
-ENDIF()
+IF (ASPECT_USE_FP_EXCEPTIONS)
+  INCLUDE(${CMAKE_SOURCE_DIR}/cmake/fpe_check.cmake)
 
-INCLUDE (CheckCXXSymbolExists)
-CHECK_CXX_SYMBOL_EXISTS("feenableexcept" "fenv.h" _HAVE_FP_EXCEPTIONS)
-IF (_HAVE_FP_EXCEPTIONS AND ASPECT_USE_FP_EXCEPTIONS)
-  MESSAGE(STATUS "Runtime floating point checks enabled.")
-  FOREACH(_source_file ${TARGET_SRC})
-    SET_PROPERTY(SOURCE ${_source_file}
-      APPEND PROPERTY COMPILE_DEFINITIONS ASPECT_USE_FP_EXCEPTIONS=1)
-  ENDFOREACH()
-ELSE()
-  IF (ASPECT_USE_FP_EXCEPTIONS)
+  IF (HAVE_FP_EXCEPTIONS)
+    MESSAGE(STATUS "Runtime floating point checks enabled.")
+    FOREACH(_source_file ${TARGET_SRC})
+      SET_PROPERTY(SOURCE ${_source_file}
+        APPEND PROPERTY COMPILE_DEFINITIONS ASPECT_USE_FP_EXCEPTIONS=1)
+    ENDFOREACH()
+  ELSE()
     SET(ASPECT_USE_FP_EXCEPTIONS OFF CACHE BOOL "" FORCE)
-    MESSAGE(STATUS "No support for feenableexcept(), disabling runtime floating point exception checks.")
+    MESSAGE(WARNING "No support for feenableexcept(), disabling runtime floating point exception checks.")
```

Comment:
This is printed as
```
[...]
-- Performing Test HAVE_FP_EXCEPTIONS - Failed
CMake Warning at CMakeLists.txt:190 (MESSAGE):
  No support for feenableexcept(), disabling runtime floating point exception
  checks.


-- Writing config into detailed.log...
[...]
```
I don't think that's particularly pretty. I would just print a regular STATUS message for this.

On the positive side, this seems to work :-) Can you update the patch?